Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary/Presenters/ImportMappingsFormPresenter.cs

```diff
@@ -23,16 +30,19 @@ public ImportMappingsFormPresenter(IImportMappingsFormView view)
 
         public void OnVisible(object sender, EventArgs e)
         {
-            if (view.SchemaConfiguration == null || !view.SchemaConfiguration.Entities.Any())
+            if (OrganizationService == null)
             {
-                ViewHelpers.ShowMessage("Please specify a schema file with atleast one entity defined.", "No entities available", MessageBoxButtons.OK, MessageBoxIcon.Error);
                 view.Close();
+                ViewHelpers.ShowMessage("Please make sure you are connected to an organisation", "No connection made", MessageBoxButtons.OK, MessageBoxIcon.Information);
                 return;
             }
-
-            view.EntityList = view.SchemaConfiguration.Entities.Select(x => x.Name).OrderBy(n => n);
+            if (new List<string>(view.EntityListDataSource).Count == 0)
```

Comment:
What is this condition for?

Reply:
It's to check whether EntityListDataSource has already been set. We only retrieve entities if it hasn't already been set

Reply:
Ah yes! I was confused about the `new List()`. Would this not work
```suggestion
            if (!view.EntityListDataSource.Any())
```